pkg/apis/v1beta1/nodeclaim.go

```diff
@@ -33,7 +33,7 @@ type NodeClaimSpec struct {
 	StartupTaints []v1.Taint `json:"startupTaints,omitempty"`
 	// Requirements are layered with GetLabels and applied to every node.
 	// +optional
-	Requirements []v1.NodeSelectorRequirement `json:"requirements,omitempty"`
+	Requirements []v1.NodeSelectorRequirement `json:"requirements,omitempty" hash:"ignore"`
 	// Resources models the resource requirements for the NodeClaim to launch
 	// +optional
 	Resources ResourceRequirements `json:"resources,omitempty"`
```

Comment:
No, this isn't the resolved value. Someone can specify this value and it _should_ imply that that's the minimum amount of requested resources that the NodeClaim needs. For now, we'll throw validation errors if this is set in the NodePool, but long-term we may want to support this dynamically, similar to requirements.

Reply:
So long term if we want to support it dynamically and currently we will not using it, shouldn't we then set  `hash:"ignore"`?

Reply:
I'm not catching the difference in setting the ignore flag now if we are planning to allow it in the future since we can assume it will always be empty due to validation?

Reply:
I was thinking it would be safer since we are never planning on statically drifting on the field. If any changes were to happen, it would require us to remember that field is not set by users due to validation. It's not obvious that `Resources` fields is not being used currently.

Reply:
Actually, that's a good point. This field would probably be dynamically updated in the future. Alright, I'm convinced. I'll update.